services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```
@@ -600,11 +601,43 @@ public VeniceHelixAdmin(
         pubSubClientsFactory.getProducerAdapterFactory(),
         metricsRepository,
         pubSubPositionTypeRegistry);
-    this.realTimeTopicSwitcher = new RealTimeTopicSwitcher(
-        topicManagerRepository.getLocalTopicManager(),
-        veniceWriterFactory,
-        commonConfig.getProps(),
-        pubSubTopicRepository);
+    if (!isParent() && commonConfig.isUseMultiRegionRealTimeTopicSwitcherEnabled()) {
+      Map<String, String> childDataCenterKafkaUrlMap = multiClusterConfigs.getChildDataCenterKafkaUrlMap();
+      String localRegionName = multiClusterConfigs.getRegionName();
+      if (!childDataCenterKafkaUrlMap.containsKey(localRegionName)) {
+        throw new VeniceException(
+            "Child data center kafka url map doesn't contain the local region: " + localRegionName);
+      }
```

Do we run into this exception due to misconfiguration?

---

Correct, since we resolve `localRegionName` and `childDataCenterKafkaUrlMap` separately in `VeniceControllerClusterConfig`. i.e. the key used to identify a region in childDataCenterKafkaUrlMap can be completely different than `localRegionName` so it's a good sanity check to fail loudly here. e.g. the map can be {dc1: xxx, dc2: yyy} while the local region can be misconfigured to something like `datacenter1`